File: Data Structures & Algorithms/duplicate-integer/submission-0.java
class Solution {
    public boolean hasDuplicate(int[] nums) {
    Set uniqueNums = new HashSet<>();

    for (int num : nums){
        if (uniqueNums.contains(num)){
            return true;
        }
        uniqueNums.add(num);
    }
    return false;
    }
}

File: Data Structures & Algorithms/subarray-sum-equals-k/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} k
     * @return {number}
     */
    subarraySum(nums, k) {

        const prefixSum = new Map();
        let res = 0, curSum = 0;
        prefixSum.set(0,1);

        for(let num of nums){
            curSum += num;
            let diff = curSum - k;
            res += prefixSum.get(diff) || 0;
            prefixSum.set(curSum, (prefixSum.get(curSum) || 0) + 1);
        }
        return res;

    }
}
